UnityProject/Assets/Scripts/Core/Threading/ThreadBehaviour.cs

```diff
@@ -82,7 +82,15 @@ private void ThreadLoop()
 			Profiler.BeginThreadProfiling("Unitystation", threadName);
 			while (running && threadMode == ThreadMode.Threaded && midTick == false)
 			{
-				RunTick();
+				try
+				{
+					RunTick();
+				}
+				catch (Exception e)
+				{
+					ThreadLogger.AddLog(e.ToString(), Category.Threading);
```

Comment:
throw as well so we notice stuff breaking

Reply:
atmospherics and electrical, both didn't have throw and those were the most likely to generate a lot of errors so

Reply:
I'm just saying it is fine to catch to log, but you should throw the exception at the end of the handling

Reply:
But the system that would have varied a lot already just had a try catch, with no throw, and the server has been fine, do you think it requires the throw being added?

Reply:
I would rather have it so we can see stuff breaking if it breaks. You first see the efect of stuff breaking then you read the logs.